active/0000-stateful-helpers.md

```diff
@@ -0,0 +1,276 @@
+- Start Date: 2015-05-17
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Stateful helpers are a class-based way to define helpers. Stateful helpers:
+
+  * Have a single return value
+  * Can store and read state
+  * Have lifecycle hooks analagous to components where appropriate. For
+    example, a stateful helper may call `recompute` at any time to generate a new
+    value.
+  * Are a superset of traditional function-based helpers. They can do more,
+    but in many cases a traditional helper is appropriate.
+
+They fill a gap in Ember's current template APIs:
+
+ |has positional params|has template|has lifecycle, instance|can control rerender
+---|---|---|---|---
+components|Yes|Yes|Yes|Yes
+function helpers|Yes|No|No|No
```

Comment:
I thought you were saying they cannot "accept a block" (aka the template) which is not correct.  The terminology of `layout` and `template` are annoying, and I am sorry that you have to thread the needle here...